packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapCircleController.m

```diff
@@ -29,6 +29,20 @@ - (instancetype)initCircleWithPosition:(CLLocationCoordinate2D)position
   return self;
 }
 
+- (instancetype)initWithCircle:(GMSCircle *)circle
+                      circleId:(NSString *)circleIdentifier
+                       mapView:(GMSMapView *)mapView
+                       options:(NSDictionary *)options {
+  self = [super init];
+  if (self) {
+    _circle = circle;
+    _mapView = mapView;
+    _circle.userData = @[ circleIdentifier ];
+    [self interpretCircleOptions:options];
+  }
+  return self;
+}
```

Comment:
These should not be two completely unrelated constructors; this is a significant maintenance problem, and violates Obj-C convention. The existing constructor should delegate to this new constructor with the `Circle` it creates.

(Same for all the other files.)

Reply:
Fixed